lib/MonadicRewrite.thy

```diff
@@ -76,12 +77,11 @@ lemma monadic_rewrite_is_refl:
   "x = y \<Longrightarrow> monadic_rewrite F E \<top> x y"
   by (simp add: monadic_rewrite_refl)
 
-lemma monadic_rewrite_refl3:
+(* precondition implies reflexivity *)
+lemma monadic_rewrite_pre_imp_refl:
   "\<lbrakk> \<And>s. P s \<Longrightarrow> f s = g s \<rbrakk> \<Longrightarrow> monadic_rewrite F E P f g"
```

Comment:
`refl` is not quite right for this one (old name, was never really right). Reflexivity requires things to be syntactically equal, i.e. `x = x`, otherwise it's just equality (`x = y`) not reflexivity. Maybe `monadic_rewrite_pre_imp_eq`?

Reply:
this one is actually used. Going to be a pain to rename and squash the changes in, but it's probably a good point.

Reply:
> For some reason the GitHub tests didn't trigger apart from the DCO. I'm very confused by this -- shouldn't be possible and the tests seem to be working fine on the previous PRs (from forks as well as internal branches). Was there anything special in the UI when you opened this PR?

Sorry, it was struggling from the beginning. I waited for ages for the whitespace test to fail, and never saw the proofs run. After I fixed the whitespace issue and pushed, I only saw DCO run and that's it.

Reply:
The rename turned out not too bad, I think.